### DISCUSSION TITLE
Configure Apache Hop to receive json from web

### DISCUSSION THREAD

**Discussion by beks6** [apache/hop#4572] [Q&A]
I asked this question on SO

https://stackoverflow.com/questions/79190748/configure-apache-hop-to-receive-json-from-web

Something seems like I'm missing here.

I looked into this "guide" and it seems simply straight forward, but when I want to send a json in the body with a POST request it doesn't work

https://hop.apache.org/manual/latest/hop-server/web-service.html

**Comment by bamaer** [comment]
I'm afraid "does not seem to consume the JSON" and "does not work" don't give us much to work with. 
Can you please share some more details? What did you try, what works, what doesn't?

**Reply by beks6** [reply]
yeah sure,

I made different approaches, also with the help of chat GPT.

I created a WebService metadata as described in the docs from hop.
My current pipeline looks like this

<img width="568" alt="Bildschirmfoto 2024-11-15 um 13 52 14" src="https://github.com/user-attachments/assets/163b3967-959a-4593-bb1f-d8114dee0636">

this approach got me the furthest so far, on the REST client transform I now get "host may not be null"

the generate rows transform is just to have a field for the REST client transform.

Maybe I'm completely on the wrong path with my solution.

My goal is to produce a json post request from my web app and send it to an on prem server which is running hop to further process the json data.

**Reply by bamaer** [reply]
so you want to do a `POST` request and get a JSON response from your API, correct? 
if that's the case, what does your REST client configuration look like?

**Reply by beks6** [reply]
No, rather the other way around. I want to send data from my web app to hop

**Reply by beks6** [reply]
After rethinking my initial approach, I decided to pull the data from the web app instead of sending it to hop.

Still would like to know if my initial idea is possible and how

**Reply by bamaer** [reply]
we have [some information in the docs](https://hop.apache.org//manual/latest/hop-server/web-service.html#_post) that explains how you can accept post requests from web service calls. does that help?

**Reply by beks6** [reply]
I tried this approach also, but wasn't able to read the body from the request.

This is now my final solution for pulling the data. Thank you so far

![Bildschirmfoto 2024-11-18 um 11 22 22](https://github.com/user-attachments/assets/102007a6-364a-46e6-ac0b-7800a26947e4)
